scheds/rust/scx_mitosis/src/bpf/mitosis.bpf.c

```diff
@@ -712,15 +764,24 @@ void BPF_STRUCT_OPS(mitosis_tick, struct task_struct *p_run)
 
 void BPF_STRUCT_OPS(mitosis_running, struct task_struct *p)
 {
+	struct cpu_ctx *cctx;
 	struct task_ctx *tctx;
 	struct cell *cell;
 
-	if (!(tctx = lookup_task_ctx(p)) || !(cell = lookup_cell(tctx->cell)))
+	if (!(tctx = lookup_task_ctx(p)) || !(cctx = lookup_cpu_ctx(-1)) ||
+	    !(cell = lookup_cell(cctx->cell)))
 		return;
 
+	/*
+        * Update both the CPU's cell and the cpu's vtime so the vtime's are
+        * comparable at dispatch time.
+        */
 	if (time_before(cell->vtime_now, p->scx.dsq_vtime))
 		cell->vtime_now = p->scx.dsq_vtime;
 
+	if (time_before(cctx->vtime_now, p->scx.dsq_vtime))
+		cctx->vtime_now = p->scx.dsq_vtime;
```

Comment:
Also, I'm not sure this is enough. This would work while the cell CPU allocation stays stable but when the assignment changes, there can be a large delta which can lead to long stalls.

Reply:
Correct, right now CPU allocation is stable - so not an issue but it will need to adjust when CPU allocation changes.